test/commands/addons/info.js

```diff
@@ -0,0 +1,63 @@
+'use strict'
+/* globals describe context it beforeEach */
+
+let fixtures = require('../../fixtures')
+let util = require('../../util')
+let cli = require('heroku-cli-util')
+let nock = require('nock')
+let cmd = require('../../../commands/addons/info')
+
+describe('addons:info', function () {
+  beforeEach(function () {
+    cli.mockConsole()
+    nock.cleanAll()
+  })
+
+  context('with add-ons', function () {
+    beforeEach(function () {
+      nock('https://api.heroku.com', {reqheaders: {'Accept-Expansion': 'addon_service,plan'}})
+        .get('/addons/www-db')
+        .reply(200, fixtures.addons['www-db'])
+      nock('https://api.heroku.com')
+        .get(`/addons/${fixtures.addons['www-db']['id']}/addon-attachments`)
+        .reply(200, [fixtures.attachments['acme-inc-www::DATABASE']])
+    })
+
+    it('prints add-ons in a table', function () {
+      return cmd.run({flags: {}, args: {addon: 'www-db'}}).then(function () {
+        util.expectOutput(cli.stdout,
+`=== www-db
+Attachments:  acme-inc-www::DATABASE
+Installed at: Invalid Date
+Owning app:   acme-inc-www
+Plan:         heroku-postgresql:hobby-dev
+Price:        free
+`)
+      })
+    })
+  })
+
+  context('with app add-ons', function () {
+    beforeEach(function () {
+      nock('https://api.heroku.com', {reqheaders: {'Accept-Expansion': 'addon_service,plan'}})
+        .get('/apps/example/addons/www-db')
+        .reply(200, fixtures.addons['www-db'])
+      nock('https://api.heroku.com')
+        .get(`/addons/${fixtures.addons['www-db']['id']}/addon-attachments`)
+        .reply(200, [fixtures.attachments['acme-inc-www::DATABASE']])
+    })
+
+    it('prints add-ons in a table', function () {
```

Comment:
@cdwort The first test is for testing that when there is no app flag that it hits `/addons/www-db` while the other test tests that when there is an app flag that it hits `/apps/example/addons/www-db`, the two tests vary by just two lines so it is admittedly hard to catch.  The resolve library handles this under the covers, this is basically verifying that the resolve library actually fixes the bug by checking the apps endpoint first when the flag is passed.  I just pushed up another commit to verify that it falls back from app addon to global addon when the app addon 404s